src/szk.rs

```diff
@@ -0,0 +1,469 @@
+            aggregator_blind.as_ref(),
+            &self.adjust_domsep_tag(DST_JOINT_RAND_PART),
+        );
+        // Encode measurement_share (currently an array of field elements) into bytes
+        // and include it in the XOF input.
+        let mut bytes = vec![0u8; T::Field::ENCODED_SIZE];
+        for field_elem in measurement_share {
+            field_elem.encode(&mut bytes)?;
+            xof.update(&bytes)
+        }
+        xof.update(nonce);
+        Ok(xof.into_seed())
+    }
+
+    #[cfg(test)]
+    fn derive_joint_rand_seed(
+        &self,
+        leader_joint_rand_part: &Seed<SEED_SIZE>,
+        helper_joint_rand_part: &Seed<SEED_SIZE>,
+    ) -> Seed<SEED_SIZE> {
+        let mut xof = P::init(
+            &[0; SEED_SIZE],
+            &self.adjust_domsep_tag(DST_JOINT_RAND_SEED),
+        );
+        xof.update(&leader_joint_rand_part.0);
+        xof.update(&helper_joint_rand_part.0);
+        xof.into_seed()
+    }
+
+    #[cfg(test)]
+    fn derive_joint_rands(
+        &self,
+        leader_joint_rand_part: &Seed<SEED_SIZE>,
+        helper_joint_rand_part: &Seed<SEED_SIZE>,
+    ) -> (Seed<SEED_SIZE>, Vec<T::Field>) {
+        let joint_rand_seed =
+            self.derive_joint_rand_seed(leader_joint_rand_part, helper_joint_rand_part);
+        let joint_rands = P::seed_stream(
+            &joint_rand_seed,
+            &self.adjust_domsep_tag(DST_JOINT_RANDOMNESS),
+            &[0],
+        )
+        .into_field_vec(self.typ.joint_rand_len());
+
+        (joint_rand_seed, joint_rands)
+    }
+
+    /// Again the binder is zero here, since the helper's agg_id and the number of proofs doesn't vary.
+    #[cfg(test)]
+    fn derive_helper_proofs_share(&self, proofs_share_seed: &Seed<SEED_SIZE>) -> Vec<T::Field> {
+        Prng::from_seed_stream(P::seed_stream(
+            proofs_share_seed,
+            &self.adjust_domsep_tag(DST_PROOF_SHARE),
+            &[0],
+        ))
+        .take(self.typ.proof_len())
+        .collect()
+    }
+
+    #[cfg(test)]
+    fn derive_query_rands(&self, verify_key: &[u8; SEED_SIZE], nonce: &[u8; 16]) -> Vec<T::Field> {
+        let mut xof = P::init(verify_key, &self.adjust_domsep_tag(DST_QUERY_RANDOMNESS));
+        xof.update(nonce);
+        xof.into_seed_stream()
+            .into_field_vec(self.typ.query_rand_len())
+    }
+
+    /*fn random_size(&self) -> usize {
+    *    if self.typ.joint_rand_len() == 0 {
+    *        // One seed for the helper's proof share and one for proving
+    *        // randomness.
+    *        2 * SEED_SIZE
+    *    } else {
+    *        // One seed for the leader, one for the helper, and one for proving
+    *        // randomness.
+    *        3 * SEED_SIZE
+    *    }
+    }*/
+
+    #[cfg(test)]
+    fn prove<'a>(
+        &'a self,
+        leader_input_share: &[T::Field],
+        helper_input_share: &[T::Field],
+        encoded_measurement: &[T::Field],
+        random_seeds: &'a [Seed<SEED_SIZE>],
```

Comment:
It seems like the leader seed is optional, based on whether the underlying FLP has joint randomness. I think this function would be more clear with arguments:
```suggestion
        prove_rand_seed: &'a Seed<SEED_SIZE>,
        helper_seed: &'a Seed<SEED_SIZE>,
        leader_seed: Option<&'a [Seed<SEED_SIZE>>,
```

Reply:
Sure